packages/@aws-cdk/aws-ecs/lib/base/pipeline-action.ts

```diff
@@ -0,0 +1,75 @@
+import codepipeline = require('@aws-cdk/aws-codepipeline-api');
+import { Repository } from '@aws-cdk/aws-ecr';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import cdk = require('@aws-cdk/cdk');
+import { BaseService } from './base-service';
+
+/**
+ * Common properties for the {@link PipelineSourceAction CodePipeline deploy Action},
+ * whether creating it directly,
+ * or through the {@link IRepository#toCodePipelineSourceAction} method.
+ */
+export interface PipelineDeployActionProps extends codepipeline.CommonActionProps {
```

Comment:
Were you planning on adding a `toCodePipelineDeployAction` method on the Service class? Because if not, the documentation on this interface seems incorrect :) (it would be cool to have that method on Service though 🙂).